### PR TITLE
create @leanjs/nuxt package

### PR DESCRIPTION
**Pull request by AlexTraher**
Introduces a new package `@leanjs/nuxt` which behaves the same way as `@leanjs/next`, providing a host component and a host provider.

There is 1 part of this that still needs some looking into, around nested routes and remotes with sub-pages as the order of the `vue-router` events vs the browser navigation events seems to be out of whack and causing a few issues. 

I thought it was worth raising this regardless as it contains almost all of the functionality.

### CODE REVIEW

File: e2e/test-subjects/shell-nuxt/app.vue [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+</template>
+
+<script setup lang="ts">
+  import { configureRuntime } from '@leanjs/core';
+  import { HostProvider } from "@leanjs/nuxt";
+  const defaultState = {
+    locale: "en",
+  };
+
+  const { createRuntime } =  configureRuntime(defaultState)({
```
Comment (alexlbr) [inline e2e/test-subjects/shell-nuxt/app.vue:21]: You can import this directly from https://github.com/leanjs/leanjs/blob/main/e2e/test-subjects/package-runtime-shared/src/index.ts

File: e2e/test-subjects/shell-next/next.config.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ const packageJsonDeps = require("./package.json").dependencies;
 const withTM = require("next-transpile-modules")([
   "@leanjs/e2e-test-subjects-package-runtime-react",
   "@leanjs/e2e-test-subjects-package-runtime-shared",
+  "@leanjs/core",
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply e2e/test-subjects/shell-next/next.config.js]: nope - resolved in 10e886d

File: e2e/test-subjects/shell-nuxt/components/Loader.vue [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<template>
+  <div>
+    Loading Micro-frontend
+  </div>
+</template>
+
+<script setup lang="ts">
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply e2e/test-subjects/shell-nuxt/components/Loader.vue]: nope - resolved in 10e886d

File: e2e/test-subjects/shell-nuxt/nuxt.config.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+export default defineNuxtConfig({
+  vite: false,
+  vue: {
+    config: {
+      productionTip: false,
+      devtools: true,
+    },
+  },
+  build: {
+    transpile: ["@leanjs/e2e-test-subjects-package-runtime-react"],
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply e2e/test-subjects/shell-nuxt/nuxt.config.ts]: we dont, I've replaced it with the general shared one in 10e886d

File: e2e/test-subjects/shell-nuxt/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    "preview": "nuxt preview"
+  },
+  "devDependencies": {
+    "nuxt3": "3.0.0-27313139.1c88580"
+  },
+  "dependencies": {
+    "@leanjs/core": "*",
+    "@leanjs/vue": "*",
+    "@leanjs/nuxt": "*",
+    "@leanjs/e2e-test-subjects-package-runtime-react": "*"
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply e2e/test-subjects/shell-nuxt/package.json:19]: same as above, removed in favour of shared one in 10e886d

File: packages/core/src/runtime/types.ts [hunk 6]
```diff
@@ -132,3 +132,5 @@ export type CreateRuntime<MyRuntime extends Runtime = Runtime> = {
 
 export type GetRuntime<MyCreateRuntime extends CreateRuntime> =
   ReturnType<MyCreateRuntime>;
+
+export const myVal = "abc";
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/core/src/runtime/types.ts]: removed in 10e886d

File: packages/nuxt/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+import VueHost from "./Host.vue";
+import VueHostProvider from "./HostProvider.vue";
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/nuxt/src/index.ts:2]: unfortunately I don't think we can :( (see the comment i left about webpack below). I tested this again and confirmed the same output - if we do this webpack doesn't produce the named exports. I'd love to be proved wrong though cos this feels like a bit of a smell.

File: packages/nuxt/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import VueHost from "./Host.vue";
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/nuxt/src/index.ts:1]: see https://github.com/leanjs/leanjs/pull/111#discussion_r922678933

File: packages/vue/src/index.ts [hunk 9]
```diff
@@ -1,2 +1,11 @@
-export * from "./createRuntimeBindings";
-export * from "./types";
+import * as runtimeBindings from "./createRuntimeBindings";
+import VueMount from "./Mount.vue";
+
+// Types are fine because tsc, not webpack handles these exports
+export * as types from "./types";
+
+// We have to do this because webpack can't properly handle re-exports
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/vue/src/index.ts:7]: yeah exactly (see https://github.com/leanjs/leanjs/pull/111#discussion_r922678933)

File: packages/vue/src/index.ts [hunk 10]
```diff
@@ -1,2 +1,11 @@
...
+export const Mount = VueMount;
+
+export const createRuntime = runtimeBindings.createRuntimeBindings;
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/vue/src/index.ts]: Whoops - clearly just an oversight when testing things, this is resolved now

File: e2e/test-subjects/shell-nuxt/app.vue (hunk 1 above)
Comment (AlexTraher) [reply e2e/test-subjects/shell-nuxt/app.vue]: done!

File: packages/nuxt/src/Host.vue [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+  if (useRoute) {
+    route = useRoute();
+  }
+
+  const basename = route?.path;
+
+  const router = useRouter();
+
+  const navigate: NavigateFunc = ({ pathname, hash, search = '' }) => {
+    router.push({ path: `${pathname}${hash}${search}` });
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/nuxt/src/Host.vue:68]: I'm not sure - I was trying to confirm exactly this when I got into the issues with nested routing. I've added a note to address this when we do nested routing

File: packages/nuxt/src/Host.vue [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+  const mountKey = url + name;
+
+  const cachedMount = mountCache.get(mountKey);
+  const mount = ref(cachedMount);
+
+
+  const error = ref<Error | null>(null);
+
+  let route;
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/nuxt/src/Host.vue]: yep - done :)

File: packages/nuxt/src/Host.vue [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+    loadModule,
+    loadScript,
+    createRemoteName,
+    deleteTrailingSlash,
+    getRemoteUrl,
+  } = CoreUtils;
+  const runtime = inject<Runtime>('runtime');
+  const props = defineProps<HostProps>();
+  const { packageName } = props.remote;
+  const origin = deleteTrailingSlash(inject<string>('origin') ?? '');
```
[the remark this answers is not in the document]
Comment (AlexTraher) [reply packages/nuxt/src/Host.vue]: I agree - resolved

**Review by alexlbr — APPROVED** [review]

**Pull request merged**